sql/item.h

```diff
@@ -873,10 +873,11 @@ class Item :public Value_source,
              COPY_STR_ITEM, FIELD_AVG_ITEM, DEFAULT_VALUE_ITEM,
              CONTEXTUALLY_TYPED_VALUE_ITEM,
              PROC_ITEM,COND_ITEM, REF_ITEM, FIELD_STD_ITEM,
-             FIELD_VARIANCE_ITEM, INSERT_VALUE_ITEM,
-             SUBSELECT_ITEM, ROW_ITEM, CACHE_ITEM, TYPE_HOLDER,
-             PARAM_ITEM, TRIGGER_FIELD_ITEM,
-             EXPR_CACHE_ITEM};
+             FIELD_REGR_AVGX_ITEM, FIELD_REGR_AVGY_ITEM, FIELD_REGR_SXX_ITEM,
```

Comment:
Let's try this a bit differently. You don't need a different field type for each individual Item.

Declare only one, `FIELD_REGR_ITEM` and write code according to that. See how it turns out. I don't think you actually need a class for each type here as you're effectively always storing and retrieving Regr_transition data structures.